ARMmbed/mbed-os
# Add parameter in tools settings to show error/warning as Link

## Description
This feature increases the comfortability of the mbed-cli within an offline IDE.
It allows the developer to jump directly to the error/warning instead of searching for the file and position.

Per default, mbed compile prints errors/warnings on output not in a link format, therefore it is not possible to click, i.e. in an IDE, on the errors/warnings and jump to the position of the file, or even know where (path) it is.
This feature generates an absolute path, because some IDEs are not parsing relatives paths correctly (i.e. VSCode).

In the linked picture, a compare between the default compile output and the newly one with the link is visualized.
[compare mbed compile output](http://www.screencast.com/t/6Aiy7brcWC7)

It is possible to activate/deactive the feature by setting this define
`PRINT_COMPILER_OUTPUT_AS_LINK = False`
in the application specific "mbed_settings.py" to true

## Status

**READY**

## Migrations

I think nothing has to be changed, it just makes the error/warning output from "mbed-cli compile" more useful in an offline IDE.

NO

## Steps to test or reproduce

Insert in the application specific "mbed_settings.py":
`PRINT_COMPILER_OUTPUT_AS_LINK = True`

Start mbed compile with:
`mbed compile -t GCC_ARM --profile release`
note: mcu (TARGET) must/should be configured in ".mbed"
